### PR TITLE
BFD-1894 Add CARIN conformant Diagnosis Codes for Sample Data in V2 

### PR DESCRIPTION
**Pull request by JeremyIR**
**JIRA Ticket:**
[BFD-1894](https://jira.cms.gov/browse/BFD-1894)

**User Story or Bug Summary:**

Add CARIN conformant Diagnosis codes to the sample data and claim type integration tests that rely on them in V2. 

---

### What Does This PR Do?

Update the Diagnosis Codeable Concepts to include codes with the ICD-10-CM system.

### What Should Reviewers Watch For?

Do the codes meet the requirements for CARIN Compliance while not causing breaking changes?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by christrotteradhoc — APPROVED** [review]
One request -- can you clarify in the PR description that this applies to the sample data and the integration tests that they are based on?

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2Test.java [hunk 1]
```diff
@@ -444,13 +444,16 @@ public void shouldHaveDiagnosesMembers() {
...
 
     DiagnosisComponent cmp5 =
         TransformerTestUtilsV2.createDiagnosis(
             // Order doesn't matter
             diag5.getSequence(),
-            new Coding("http://hl7.org/fhir/sid/icd-10", "R1122", null),
+            new Coding(
+                "http://hl7.org/fhir/sid/icd-10",
+                "B20",
+                "HUMAN IMMUNODEFICIENCY VIRUS �HIV� DISEASE"),
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2Test.java:456]: Whats with the weird encoding here?

**Review by lsmitchell — APPROVED** [review]
+10 for RAT-BITE FEVERS

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2Test.java (hunk 1 above)
Comment (JeremyIR) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/OutpatientClaimTransformerV2Test.java:456]: Thats unfortunately what our transformer code does with these brackets that are provided with the display for HIV.

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/InpatientClaimTransformerV2Test.java [hunk 2]
```diff
@@ -688,15 +688,21 @@ public void shouldHaveDiagnosesList() {
...
         TransformerTestUtilsV2.createDiagnosis(
             // Order doesn't matter
             diag1.getSequence(),
             List.of(
-                new Coding("http://hl7.org/fhir/sid/icd-10-cm", "R4444", null),
-                new Coding("http://hl7.org/fhir/sid/icd-10", "R4444", null)),
+                new Coding(
+                    "http://hl7.org/fhir/sid/icd-10-cm",
+                    "A05",
+                    "\"OTH BACTERIAL FOODBORNE INTOXICATIONS, NEC\""),
```
Comment (mackec1) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/InpatientClaimTransformerV2Test.java:701]: Is this really a string that include the double-quotes?

File: apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v1/eobByPatientIdAll.json [hunk 3]
```diff
@@ -2041,7 +2053,8 @@
         "diagnosisCodeableConcept" : {
           "coding" : [ {
             "system" : "http://hl7.org/fhir/sid/icd-10",
-            "code" : "R4444"
+            "code" : "A05",
+            "display" : "\"OTH BACTERIAL FOODBORNE INTOXICATIONS, NEC\""
```
Comment (mackec1) [inline apps/bfd-server/bfd-server-war/src/test/resources/endpoint-responses/v1/eobByPatientIdAll.json:2057]: I'm sure we get these descriptions from external sources, but WTF....why they feel compelled to include quotes within a description is beyond me.

**Review by mackec1 — COMMENTED** [review]
Couple of things here, but I suspect they are the by-product of the code(s) used in the tests. Is there a possibility of using code(s) that don't have the questionable description (double-quotes, invisible embedded chars)? Then if/when the provider actually cleans up their crappy data, our tests won't break.

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/InpatientClaimTransformerV2Test.java (hunk 2 above)
Comment (JeremyIR) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/InpatientClaimTransformerV2Test.java]: Just removed all weirdly encoded diagnoses.

**Review by christrotteradhoc — DISMISSED** [review]

**Review by mackec1 — APPROVED** [review]
LOTS of good diseases in test data....remind me not to get sick!

**Review by christrotteradhoc — APPROVED** [review]

**Pull request merged**